scripts/dts/extract/clocks.py

```diff
@@ -5,30 +5,73 @@
 #
 
 from extract.globals import *
+from extract.edts import *
 from extract.directive import DTDirective
 
 ##
 # @brief Manage clocks related directives.
 #
 # Handles:
 # - clocks
-# directives.
+#
+# Generates in EDTS:
+#
+# Clock consumer
+# --------------
+# - clocks/<index>/<cell-name> : <cell-value>
+#                  (cell-name from cell-names of provider)
 #
 class DTClocks(DTDirective):
 
     def __init__(self):
         pass
 
+    ##
+    # @brief Insert clock cells into EDTS
+    #
+    # @param clock_consumer
+    # @param yaml
+    # @param clock_provider
+    # @param clock_cells
+    # @param property_path_templ "xxx/yyy/{}"
+    #
+    def _edts_insert_clock_cells(self, clock_consumer_node_address, yaml,
+                                 clock_provider_node_address, clock_cells,
+                                 property_path_templ):
+        if len(clock_cells) == 0:
+            return
+        clock_consumer_device_id = edts_device_id(clock_consumer_node_address)
+        clock_provider_device_id = edts_device_id(clock_provider_node_address)
+        clock_provider = reduced[clock_provider_node_address]
+        clock_provider_compat = get_compat(clock_provider_node_address)
+        clock_provider_bindings = yaml[clock_provider_compat]
+        clock_nr_cells = int(clock_provider['props'].get('#clock-cells', 0))
+        clock_cells_names = clock_provider_bindings.get(
```

Comment:
What happens if clock_nr_cells is 4 or 10?

Reply:
CELLx will be created as name - see the following code where the length of the clock_cell_names list is checked while generating clock_cell_name.

The default list could be shrinked to ['ID', ] as the other cell names will automatically be created to 'CELLx' if no name is given.

Reply:
sounds good, that would be cleaner.